torch/_inductor/scheduler.py

```diff
@@ -3945,11 +3959,162 @@ def can_buffer_be_removed_through_fusion(
             and name not in self.mutation_real_name
         )
 
+    def should_partition(self, node: BaseSchedulerNode) -> bool:
+        if not node.is_gpu():
+            return True
+
+        if isinstance(node.node, ir.DeviceCopy):
+            return True
+
+        if isinstance(node.node, ir.Conditional):
+            return True
+
+        if getattr(node.node, "unbacked_bindings", None):
+            return True
+
+        return False
+
+    def get_name_to_nodes(self) -> ir.PartitionInputType:
+        name_to_node: ir.PartitionInputType = {}
+        name_to_node.update(V.graph.graph_inputs)
+
+        for node in self.nodes:
+            for name, scheduler_buffer in node.outputs_by_name.items():
+                name_to_node[name] = scheduler_buffer.node
+
+        return name_to_node
+
+    def get_graph_partition_signature(
+        self, partitions: List[PartitionType]
+    ) -> Tuple[List[PartitionInputMetadataType], List[ir.PartitionOutputType],]:
```

Comment:
nit: per recent pep we changed all the `list[T], tuple[T]` -> `list[T], tuple[T]`